km/km_hcalls.c

```diff
@@ -2209,17 +2209,30 @@ static km_hc_ret_t sched_yield_hcall(void* vcpu, int hc, km_hc_args_t* arg)
 
 static km_hc_ret_t inotify_init_hcall(void* vcpu, int hc, km_hc_args_t* arg)
 {
-   // int inotify_init(void);
-   km_warnx("Unsupported call inotify_init");
-   arg->hc_ret = -ENOTSUP;
+   arg->hc_ret = km_fs_inotify_init(vcpu);
```

Comment:
Just curious why you didn't turn this into a call to km_fs_inotify_init1() and set the flags argument to zero.
What you have is fine though.